### PR TITLE
Add missing types: FileType, SourceType, ReaderType

### LINKED ISSUE #11: Add support to FileType
Implement the `extractHeaders` and the `parsedCSV` in the Parser object so that this type is handled in those functions

### LINKED ISSUE #14: Add support to ReaderType
Implement the `extractHeaders` and the `parsedCSV` in the Parser object so that this type is handled in those functions

### LINKED ISSUE #16: Add support to SourceType
Implement the `extractHeaders` and the `parsedCSV` in the Parser object so that this type is handled in those functions

### PR DESCRIPTION
**Pull request by ricardochaves9**
Added
Add support to FileType #11
Support to ReaderType #14
Support to SourceType #16

### CODE REVIEW

File: src/main/scala/parsing/Parser.scala [hunk 1]
```diff
@@ -10,13 +12,25 @@ object Parser extends Encoding {
...
-      case FileType(input)     => ???
       case IteratorType(input) => ???
       case IterableType(input) => ???
-      case ReaderType(input)   => ???
-      case SeqType(input) =>
-        val parsed: Seq[Either[String, String]] = input.map(parsingLogic)
+      case ReaderType(input)   =>
+        val source =  Source.fromString(input.toString)
+        val sourceLines = source.getLines()
+        val parsed = sourceLines.toList.map(parsingLogic)
```
Comment (PedroCorreiaLuis) [inline src/main/scala/parsing/Parser.scala]: missing types

File: src/main/scala/parsing/Parser.scala [hunk 2]
```diff
@@ -28,19 +42,50 @@ object Parser extends Encoding {
...
   private val extractHeaders: ParserInput => ParserInput = { parserInput =>
     {
       if (parserInput.headers.isEmpty) {
         parserInput.in match {
-          case FileType(input)     => ???
+          case FileType(input) =>
+            val source= Source.fromFile(input)
+            val sourceLines: Iterator[String] = source.getLines()
+            val headers: List[EncodeType] = sourceLines.take(1).toList
+            val lines: Iterator[String] = sourceLines
```
Comment (PedroCorreiaLuis) [inline src/main/scala/parsing/Parser.scala]: types

File: src/main/scala/parsing/Parser.scala [hunk 3]
```diff
@@ -28,19 +42,50 @@ object Parser extends Encoding {
...
+                csvDefinition = parserInput.csvDefinition,
+                headers = headers)
           case IteratorType(input) => ???
           case IterableType(input) => ???
-          case ReaderType(input)   => ???
+          case ReaderType(input) =>
+            val source = Source.fromString(input.toString)
+            val sourceLines = source.getLines()
+            val headers = sourceLines.take(1).toList
+            val lines = sourceLines
```
Comment (PedroCorreiaLuis) [inline src/main/scala/parsing/Parser.scala]: types

**Review by PedroCorreiaLuis — COMMENTED** [review]
We need to start to add tests

File: src/main/scala/parsing/Parser.scala (hunk 1 above)
Comment (ricardochaves9) [reply src/main/scala/parsing/Parser.scala]: done

File: src/main/scala/parsing/Parser.scala (hunk 2 above)
Comment (ricardochaves9) [reply src/main/scala/parsing/Parser.scala]: done

File: src/main/scala/parsing/Parser.scala (hunk 3 above)
Comment (ricardochaves9) [reply src/main/scala/parsing/Parser.scala]: done

**Review by PedroCorreiaLuis — APPROVED** [review]

**Pull request merged**